src/bin/taker.rs

```diff
@@ -168,6 +176,39 @@ fn main() {
             let address = taker.get_wallet_mut().get_next_external_address().unwrap();
             println!("{:?}", address);
         }
+        Commands::SendToAddress { address, amount } => {
+            // Assuming fee=fee_rate
+            // TODO: find a way to derive fee.
```

Comment:
No fee and fee_rate is different. Doing fee_rate is difficult, because `fee = fee_rate * tx_weight`. And we don't know the `tx_weight` before signing the transaction. But we need the `fee` to create the outputs before signing the tx.

So it becomes a chicken and egg style problem.

One way to solve this is to pre-estimate the `tx_weight`, which is possible because its always single-sig p2wpkh type, which has a constant size of tx input. 

This will get solved by BDK as it internally knows how to estimate input size from the descriptor. 

For our purpose now having just the `fee` is will work.

Reply:
> IMO, it seems irrational from user perspective to give fee as argument while calling send-to-addresss .

Yes but the other way is worse. Telling its a fee-rate in name while its not. That can be even called as a bug.  

The answer here is to fix the problem and add fee-rate handling logic inside `spend_from_wallet`. Not keep pretending fee=fee_rate. 

Also it's a pretty deep API that no "user" is going to use anytime soon. For the devs we can add a comment saying, why we take the fee and its a limitation of the API, accept that we are stupid, and assure them that it will eventually be fixed with BDK. :smile: